src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -1634,6 +1638,40 @@ namespace cryptonote
     return true;
   }
   //-----------------------------------------------------------------------------------------------
+  bool core::check_service_node_time()
+  {
+  
+    crypto::public_key pubkey = m_service_node_list.get_random_pubkey();
+    crypto::x25519_public_key x_pkey{0};
+    m_service_node_list.access_proof(pubkey, [&](auto &proof) { x_pkey = proof.pubkey_x25519; });
+    if (x_pkey) {
+      m_lmq->request(
+          std::to_string(x_pkey),
+          "quorum.timestamp",
+          [&](bool success, std::vector<std::string> data) {
+            m_service_node_list.record_timestamp_participation(pubkey, success);
```

Comment:
`pubkey` is a local variable passed by reference into the lambda, should be a passed by value into the lambda instead.